internal/pkg/repository/repository.go

```diff
@@ -28,28 +30,24 @@ type Registry interface {
 type Repository struct {
 	name     string
 	registry Registry
-
-	uri string
 }
 
 // New instantiates a new Repository.
-func New(name string, registry Registry) (*Repository, error) {
-	uri, err := registry.RepositoryURI(name)
-	if err != nil {
-		return nil, fmt.Errorf("get repository URI: %w", err)
-	}
-
+func New(name string, sess *session.Session) (*Repository, error) {
```

Comment:
can we take a `Registry` as input? I don't think any other package besides `cli` should actually initialize clients so that we do dependency injection and reuse clients

```suggestion
func New(name string, registry Registry) (*Repository, error) {
```

Reply:
I actually was in two minds about this and changed at the last minute. It's because if I want the `cli` package to avoid directly using any lower level `aws/` packages, and control by passing sessions instead. What do you think?

Reply:
That also makes sense, but I think we should establish a convention that encourages more taking interfaces as inputs if you're not the `cli` pkg so that engineers aren't confused how to create constructors.

It's okay if the `cli` package is a bit verbose, but the majority of the logic will be just about initializing clients

Reply:
oke done